core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -137,6 +138,22 @@ trait EntityDecoderInstances {
     EntityDecoder.decodeBy(MediaRange.`*/*`)(collectBinary)
   }
 
+  implicit def sourceDecoder[A](implicit W: EntityDecoder[A]): EntityDecoder[Process[Task,A]] =
+    new EntityDecoder[Process[Task,A]] {
+      override def decode(msg: Message): DecodeResult[Process[Task,A]] = {
+        val result = msg.body.flatMap{ byteVector =>
+          val fakeMessage = Request(body = Process.emit(byteVector))
```

Comment:
Well, the signature of EntityDecoder feels okay for when it applies to a whole HTTP message. But it does not lend itself well to being used recursively like it would need to be to expose HTTP chunked encoding.

This why I think it might make sense to not expose this to the HTTP client. I am not even sure there is a use case for this, I just assumed that if it was being streamed from the server then it would be useful to receive it as a stream from the client. But I think chunked encoding is meant to be somewhat of an optimization for the server, as opposed to a way to expose streaming to the client. If the client needs streaming, it should use websockets.

Am I making any sense?